document/chapters/preliminaries.tex

```diff
@@ -1,3 +1,279 @@
 % !TEX root = ../main.tex
 \chapter{Preliminaries}
 \label{chapter:preliminaries}
+
+Rust is a systems programming language focused on speed, memory safety and concurrency. \footnote{\url{https://www.rust-lang.org/}}  It intends to offer both performance similar to C++ and memory safety similar to Haskell. Rust is a compiled language combining imperative and functional programming features. The focus on performance makes Rust an ideal candidate for writing operative systems, databases, compilers, and other high-performance software without worrying about manual memory allocation. However, Rust's safety and high-level abstractions has encouraged its usage in backend and even frontend development. 
```

Comment:
Rust is a systems programming language focused on speed, memory safety and concurrency.\footnote{\url{https://www.rust-lang.org/}}  It intends to offer both performance similar to C++ and memory safety similar to Haskell. Rust is a compiled language combining imperative and functional programming features. The focus on performance makes Rust an ideal candidate for writing operative systems, databases, compilers, and other high-performance software without worrying about manual memory allocation. However, Rust's safety and high-level abstractions has encouraged its usage in backend and even frontend development.